server/src/main/java/io/druid/client/HttpServerInventoryView.java

```diff
@@ -121,10 +122,11 @@ public HttpServerInventoryView(
       final @Global HttpClient httpClient,
       final DruidNodeDiscoveryProvider druidNodeDiscoveryProvider,
       final Predicate<Pair<DruidServerMetadata, DataSegment>> defaultFilter,
-      final HttpServerInventoryViewConfig config
-  )
+      final HttpServerInventoryViewConfig config,
+      final AuthenticatorHttpClientWrapper authenticatorHttpClientWrapper
```

Comment:
Is it possible to adjust guice bindings so that httpClient injected is already wrapped? then this kind of change wouldn't be needed in all the places.

Reply:
It is, and I had implemented it that way initially, but changed it after a discussion with @gianm, we felt it would be better to have the call sites explicitly request escalation rather than always wrapping the client, that it would be better to avoid putting auth credentials into requests unless truly necessary

Reply:
given that all the places need a escalated client, I would make existing clients escalated and if there is a real use case (now or in future) for unescalated client then let us bing `@unescalatedGlobal` etc clients in guice that could be injected in those specific places.

@jon-wei @gianm does that sound ok ?

Reply:
I think for "fail-secure" coding it's better for the default client to be unescalated. That way you won't accidentally get an escalated client in a path that isn't checking auth some other way -- which would be a security hole.

Reply:
@himanshug @nishantmonu51  I'll update this patch to add @EscalatedGlobal and @EscalatedClient annotations for HttpClient, I agree with @gianm on keeping unescalated client as the default, how does that sound?

Reply:
my main concern was having that repeated wrapper code everywhere, that gets solved either way. so, I'm alright.

Does this mean `@Global` and `@Router` annotations would be used to get unescalated clients? We should probably note this in release notes as there might be extensions out there using `@Global` etc to get httpClient and might need escalated clients really.